keyboards/planck/keymaps/a2b/keymap.c

```diff
@@ -0,0 +1,866 @@
+  state = update_tri_layer_state(state, _LOWER, _ALTGR, _LOWERAG);
+  state = update_tri_layer_state(state, _RAISE, _ALTGR, _RAISEAG);
+  return state;
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+  case BEPO: // base layer, bépo layout using azerty driver on windows
+      if (record->event.pressed) {
+	set_single_persistent_default_layer(_BEPO);
+ 	layer_clear();
+      }
+      return false;
+      break;
+  case AZERTY: // base layer, bépo layout using azerty driver on windows
+      if (record->event.pressed) {
+        set_single_persistent_default_layer(_AZERTY);
+      }
+      return false;
+      break;
+  /* case AZ_LOWER: // turn on layer _SHIFT with modifier Left Shift held */
+  /*     if (record->event.pressed) { */
+  /*       layer_on(_LOWER); */
+  /*       layer_on(_AZ_LOWER); */
+  /*     } */
+  /*     else { */
+  /*       layer_off(_AZ_LOWER); */
+  /*       layer_off(_LOWER); */
+  /*     } */
+  /*     return false; */
+  /*     break; */
+  /* case AZ_RAISE: // turn on layer _SHIFT with modifier Left Shift held */
+  /*     if (record->event.pressed) { */
+  /*       layer_on(_RAISE); */
+  /*       layer_on(_AZ_RAISE); */
+  /*     } */
+  /*     else { */
+  /*       layer_off(_AZ_RAISE); */
+  /*       layer_off(_RAISE); */
+  /*     } */
+  /*     return false; */
+  /*     break; */
+  case SHIFT: // turn on layer _SHIFT with modifier Left Shift held
+      if (record->event.pressed) {
+	register_code(KC_LSFT);
+        layer_on(_SHIFT);
+      }
+      else {
+        layer_off(_SHIFT);
+	unregister_code(KC_LSFT);
+      }
+      return false;
+      break;
+  case ALTGR: // turn on layer _ALTGR with modifier AltGr held
+      if (record->event.pressed) {
+	register_code(KC_ALGR);
+        layer_on(_ALTGR);
+      }
+      else {
+        layer_off(_ALTGR);
+	unregister_code(KC_ALGR);
+      }
+      return false;
+      break;
+  case ASFT_I: // personnal Mod-Tap for _SHIFT layer and KC_LSFT on
+    if (record->event.pressed) { // key is pressed, set timer
+      is_sft_i_pressed = true;
+      sft_i_timer = timer_read();
+    }
+    else { // key is releasd, see if it was tapped (send FR_I)
+      // or held (in this case, layer _SHIFT was turned on and Left Shift held)
+      is_sft_i_pressed = false;
+      if (!is_sft_i_active) {
+	tap_code16(FR_I);
+      }
+      else {
+	layer_off(_SHIFT);
+	unregister_code(KC_LSFT);
+	is_sft_i_active = false;
+      }
+    }
+    break;
+  case ASFT_S: // personnal Mod-Tap for _SHIFT layer and KC_LSFT on
+    // see SFT_I for details
+    if (record->event.pressed) {
+      is_sft_s_pressed = true;
+      sft_s_timer = timer_read();
+    }
+    else {
+      is_sft_s_pressed = false;
+      if (!is_sft_s_active) {
+	tap_code16(FR_S);
+      }
+      else {
+	layer_off(_SHIFT);
+	unregister_code(KC_LSFT);
+	is_sft_s_active = false;
+      }
+    }
+    break;
+  case UC_REE...UC_GEQL:
```

Comment:
This looks like you're just re-implementing the unicode map feature?

Reply:
Yes, that is exactly the case since unicode map feature does not work with layouts with numbers on shift or I was not able to do it porperly.